six/modules/c++/six.sicd/source/Grid.cpp

```diff
@@ -84,4 +480,812 @@ bool Grid::operator==(const Grid& rhs) const
+    valid = col->validate(imageData, log) && valid;
+    return valid;
+}
+
+void Grid::fillDerivedFields(
+        const CollectionInformation& collectionInformation,
+        const ImageData& imageData,
+        const SCPCOA& scpcoa)
+{
+    if (!Init::isUndefined<double>(scpcoa.scpTime) &&
+        collectionInformation.radarMode == RadarModeType::SPOTLIGHT &&
+        Init::isUndefined<Poly2D>(timeCOAPoly))
+    {
+        timeCOAPoly = Poly2D(0, 0);
+        timeCOAPoly[0][0] = scpcoa.scpTime;
+    }
+
+    row->fillDerivedFields(imageData);
+    col->fillDerivedFields(imageData);
+}
+
+void Grid::fillDerivedFields(const RMA& rma, const Vector3& scp,
+        const PolyXYZ& arpPoly)
+{
+    if (rma.rmat.get())
+    {
+        fillDerivedFields(*rma.rmat, scp);
+    }
+    else if (rma.rmcr.get())
+    {
+        fillDerivedFields(*rma.rmcr, scp);
+    }
+    else if (rma.inca.get())
+    {
+        fillDerivedFields(*rma.inca, scp, arpPoly);
+    }
+}
+
+void Grid::fillDerivedFields(const RMAT& rmat, const Vector3& scp)
+{
+    // Row/Col.UnitVector and Derived fields
+    if (Init::isUndefined<Vector3>(row->unitVector) &&
+        Init::isUndefined<Vector3>(col->unitVector))
+    {
+        row->unitVector = derivedRowUnitVector(rmat, scp);
+        col->unitVector = derivedColUnitVector(rmat, scp);
+    }
+}
+
+void Grid::fillDerivedFields(const RMCR& rmcr, const Vector3& scp)
+{
+    // Row/Col.UnitVector and Derived fields
+    if (Init::isUndefined<Vector3>(row->unitVector) &&
+        Init::isUndefined<Vector3>(col->unitVector))
+    {
+        row->unitVector = derivedRowUnitVector(rmcr, scp);
+        col->unitVector = derivedColUnitVector(rmcr, scp);
+    }
+}
+
+void Grid::fillDerivedFields(const INCA& inca, const Vector3& scp,
+        const PolyXYZ& arpPoly)
+{
+    if (!Init::isUndefined<Poly1D>(inca.timeCAPoly) &&
+        !Init::isUndefined<PolyXYZ>(arpPoly) &&
+        Init::isUndefined<Vector3>(row->unitVector) &&
+        Init::isUndefined<Vector3>(col->unitVector))
+    {
+        row->unitVector = derivedRowUnitVector(inca, scp, arpPoly);
+        col->unitVector = derivedColUnitVector(inca, scp, arpPoly);
+    }
+
+    if (Init::isUndefined<double>(col->kCenter))
+    {
+        col->kCenter = 0;
+    }
+
+    if (!Init::isUndefined<double>(inca.freqZero) &&
+        Init::isUndefined<double>(row->kCenter))
+    {
+        row->kCenter = derivedRowKCenter(inca);
+    }
+}
+
+void Grid::fillDerivedFields(const RgAzComp& rgAzComp,
+        const GeoData& geoData,
+        const SCPCOA& scpcoa,
+        double fc)
+{
+    const Vector3& scp = geoData.scp.ecf;
+
+    if (imagePlane == ComplexImagePlaneType::NOT_SET)
+    {
+        imagePlane = ComplexImagePlaneType::SLANT;
+    }
+    if (imagePlane == ComplexImageGridType::NOT_SET)
+    {
+        type = ComplexImageGridType::RGAZIM;
+    }
+
+    if (Init::isUndefined<Vector3>(row->unitVector))

```

Comment:
Do you really need the explicit `<Vector3>` here and in the line below?  I would've thought the compiler would deduce it.